guide/slash-commands/autocomplete.md

```diff
@@ -22,27 +22,46 @@ const data = new SlashCommandBuilder()
+To handle an <DocsLink path="class/AutocompleteInteraction"/>, use the <DocsLink path="class/BaseInteraction?scrollTo=isAutocomplete"/> type guard to make sure the interaction instance is an autocomplete interaction. You can do this in a separate `interactionCreate` listener:
 
 <!-- eslint-skip -->
 
 ```js {1,4}
 client.on('interactionCreate', interaction => {
 	if (!interaction.isAutocomplete()) return;
+	// do autocomplete handling
+});
+```
+
+Or alternatively, by making a small change to your existing [Command handler](/creating-your-bot/command-handling.md).
+
+```js {2-6}
+client.on('interactionCreate', interaction => {
+	if (interaction.isChatInputCommand()) {
+		// do command handling
+	} else if (interaction.isAutocomplete()) {
+		// do autocomplete handling
+	}
 });
 ```
 
-The <DocsLink path="class/AutocompleteInteraction"/> class provides the <DocsLink path="class/AutocompleteInteraction?scrollTo=respond"/> method to send a response.
+:::tip
+You might have already moved this code to `events/interactionCreate.js` if you followed our [Event handling](/creating-your-bot/event-handling.md) guide too.
+:::
 
 ### Sending results
 
-Using <DocsLink path="class/AutocompleteInteraction?scrollTo=respond" /> you can submit an array of <DocsLink path="typedef/ApplicationCommandOptionChoiceData" /> objects. Passing an empty array will show "No options match your search" for the user.
+The <DocsLink path="class/AutocompleteInteraction"/> class provides the <DocsLink path="class/AutocompleteInteraction?scrollTo=respond"/> method to send a response. Using this, you can submit an array of <DocsLink path="typedef/ApplicationCommandOptionChoiceData" /> objects for the user to choose from. Passing an empty array will show "No options match your search" for the user.
+
+::: warning
+Unlike static choices, autocompletion suggestions are *not* enforced, and users may still enter free text.
+:::
+
+The <DocsLink path="class/CommandInteractionOptionResolver?scrollTo=getFocused" /> method returns the currently focused option's value, which can be used to applying filtering to the choices presented. For example, to only display options starting with the focused value you can use the `Array#filter()` method, then using `Array#map()`, you can transform the array into an array of <DocsLink path="typedef/ApplicationCommandOptionChoiceData" /> objects.
 
-The <DocsLink path="class/CommandInteractionOptionResolver?scrollTo=getFocused" /> method returns the currently focused option's value. This value can be used to filter the choices presented. For example, to only display options starting with the focused value you can use the `Array#filter()` method, then using `Array#map()`, you can transform the array into an array of <DocsLink path="typedef/ApplicationCommandOptionChoiceData" /> objects.
+The example below shows how this might be applied to a conceptual version of our `guide` command to determine the closest topic to the search input:
```

Comment:
```suggestion
The example below shows how this might be applied to a conceptual version of the `guide` command to determine the closest topic to the search input:
```